test-integration/Microsoft.Azure.WebJobs.Extensions.OpenApi.Document.Tests/Post_AppicationJson_Decimal_Tests.cs

```diff
@@ -0,0 +1,114 @@
+        {
+            var content = this._doc["paths"][path][operationType]["requestBody"]["content"];
+
+            content[contentType].Should().NotBeNull();
+        }
+
+        [DataTestMethod]
+        [DataRow("/post-applicationjson-decimal", "post", "text/plain", "number", "decimal")]
+        public void Given_OpenApiDocument_Then_It_Should_Return_OperationRequestBodyContentTypeSchema(string path, string operationType, string contentType, string propertyType, string propertyFormat)
+        {
+            var content = this._doc["paths"][path][operationType]["requestBody"]["content"];
+
+            var value = content[contentType]["schema"];
+
+            value.Should().NotBeNull();
+            value.Value<string>("type").Should().Be(propertyType);
+            value.Value<string>("format").Should().Be(propertyType);
+        }
+
+        [DataTestMethod]
+        [DataRow("/post-applicationjson-decimal", "post", "200")]
+        public void Given_OpenApiDocument_Then_It_Should_Return_OperationResponse(string path, string operationType, string responseCode)
+        {
+            var responses = this._doc["paths"][path][operationType]["responses"];
+
+            responses[responseCode].Should().NotBeNull();
+        }
+
+        [DataTestMethod]
+        [DataRow("/post-applicationjson-decimal", "post", "200", "application/json")]
+        public void Given_OpenApiDocument_Then_It_Should_Return_OperationResponseContentType(string path, string operationType, string responseCode, string contentType)
+        {
+            var content = this._doc["paths"][path][operationType]["responses"][responseCode]["content"];
+
+            content[contentType].Should().NotBeNull();
+        }
+
+        [DataTestMethod]
+        [DataRow("/post-applicationjson-decimal", "post", "200", "application/json", "decimalObjectModel")]
+        public void Given_OpenApiDocument_Then_It_Should_Return_OperationResponseContentTypeSchema(string path, string operationType, string responseCode, string contentType, string reference)
+        {
+            var content = this._doc["paths"][path][operationType]["responses"][responseCode]["content"];
+
+            var @ref = content[contentType]["schema"]["$ref"];
+
+            @ref.Value<string>().Should().Be($"#/components/schemas/{reference}");
+        }
+
+        [DataTestMethod]
+        [DataRow("decimaleObjectModel", "object")]
+        public void Given_OpenApiDocument_Then_It_Should_Return_ComponentSchema(string @ref, string refType)
+        {
+            var schemas = this._doc["components"]["schemas"];
+
+            var schema = schemas[@ref];
+
+            schema.Should().NotBeNull();
+            schema.Value<string>("type").Should().Be(refType);
+        }
+
+        [DataTestMethod]
+        [DataRow("decimalObjectModel", "decimalValue", "number", "decimal")]
```

Comment:
Same problem with "decimalObjectModel". your app wrote your model type name "decimalTypeObjectModel". Please change this.

Reply:
thanks i fixed it